Fix AnyPattern nullable bug

aeoncl opened the pull request:
<!-- What changes are being made? (What feature/bug is being fixed here?) -->

**What**: Changing how newBaseOn & listOf of the AnyPattern handle the presence of NullPattern in their PatternList

**Why**: The use of the method single() on the PatternList throwed an Exception when trying to generate a nullable array

**How**: Adding checks and excluding the NullablePattern from the selection in specific cases

**Checklist**:

- [NA] Documentation added to the README.md OR link to PR on https://github.com/specmatic/specmatic-documentation
- [X] Tests
- [NA] Sonar Quality Gate

<!-- Kindly link the issue that this PR will be addressing -->
**Issue ID**:
Closes: #742

<!-- feel free to add additional comments -->

Commit:
Fix AnyPattern nullable bug

joelrosario reviewed (changes requested):
Need a test for the new functionality.

Commit:
Add tests for  AnyPattern nullable bug

Commit:
Removed bad import

joelrosario committed:
Used "first" instead of "single" in AnyPattern.listOf

listOf is intended to wrap the provided values in a list based on the
type of values. For example, a string would be wrapped in a
JSONArrayValue object. An XMLNode object would be wrapped in another
XMLNode object.

We can safely assume that all the types in an AnyPattern at any given
point due to the way that the object is constructed would consist of
values of the same type (all the values would be either JSON or XML).

Therefore, instead of using single (which can break given an enum type
 with multiple options), we use first to identify the element in the
 list of values that shall provide a wrapping list object.

This commit also includes a refactored version of the new tests.

aeoncl committed:
Empty-Commit for correct user

joelrosario reviewed (approved):
Looks good to me, thanks @aeoncl!
